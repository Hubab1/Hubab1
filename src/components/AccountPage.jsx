import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Formik } from 'formik';
import * as Yup from 'yup';
import Grid from '@material-ui/core/Grid';
import { KeyboardDatePicker } from '@material-ui/pickers';

import { updateApplicant } from 'reducers/applicant';
import captureRoute from 'app/captureRoute';
import { H1, formContent } from 'assets/styles';
import FormTextInput from 'components/common/FormTextInput/FormTextInput';
import PhoneNumberInput from 'components/common/PhoneNumberInput';
import GenericFormError from 'components/common/GenericFormError';
import ActionButton from 'components/common/ActionButton/ActionButton';
import { ROUTES } from 'app/constants';
import { allValuesSet } from 'utils/formik';
import { serializeDate, parseDateISOString } from 'utils/misc';	


export class AccountPage extends React.Component {
    state = {errors: null}

    get initialValues () {
        const applicant = this.props.applicant;
        let birthday = applicant.birthday;
        if (birthday) {
            birthday = parseDateISOString(birthday);
        }
        return {
            first_name: applicant.first_name,
            last_name: applicant.last_name,
            phone_number: applicant.phone_number,
            email: applicant.email,
            birthday
        }
    }

    onSubmit = (values, { setSubmitting, setErrors }) => {
        const stateUpdate = Object.assign({}, values);
        stateUpdate.birthday = serializeDate(stateUpdate.birthday);
        this.props.updateApplicant(stateUpdate, stateUpdate).then((res) => {
            if (res.errors) {
                setErrors(res.errors);
            } else {
                
            }
            setSubmitting(false);
        });
    }

    render () {
        return (
            <>
                <H1>Your Account Details</H1>
                <Formik
                    initialValues={this.initialValues}
                    validationSchema={Yup.object().shape({
                        first_name: Yup.string().required('First Name is required'),
                        last_name: Yup.string().required('Last Name is required'),
                        phone_number: Yup.string()
                            .required('Phone Number is required')
                            .matches(/^\(\d{3}\)\s\d{3}-\d{4}/, 'Must be a valid US phone number'),
                        email: Yup.string()
                            .email()
                            .required('Email is required'),
                        birthday: Yup.string()	
                            .required('required'),
                    })}
                    onSubmit={this.onSubmit}
                >
                    {({
                        values,
                        errors,
                        handleChange,
                        submitCount,
                        handleBlur,
                        handleSubmit,
                        isSubmitting,
                        touched,
                        setFieldValue,
                    }) => (
                        <form onSubmit={handleSubmit} autoComplete="off">
                            <div className={formContent}>
                                { this.state.errors && <GenericFormError errors={this.state.errors}/> }
                                <Grid container spacing={1}>
                                    <Grid item xs={12}>
                                        <FormTextInput
                                            label="First Name"
                                            name="first_name"
                                            submitted={submitCount > 0}
                                            handleChange={handleChange}
                                            handleBlur={handleBlur}
                                            error={errors.first_name}
                                            value={values.first_name}
                                        />
                                    </Grid>
                                    <Grid item xs={12}>
                                        <FormTextInput
                                            label="Last Name"
                                            name="last_name"
                                            submitted={submitCount > 0}
                                            handleChange={handleChange}
                                            handleBlur={handleBlur}
                                            error={errors.last_name}
                                            value={values.last_name}
                                        />
                                    </Grid>
                                    <Grid item xs={12}>
                                        <FormTextInput
                                            label="Email"
                                            name="email"
                                            submitted={submitCount > 0}
                                            handleChange={handleChange}
                                            handleBlur={handleBlur}
                                            error={errors.email}
                                            value={values.email}
                                        />
                                    </Grid>
                                    <Grid item xs={6}>
                                        <PhoneNumberInput 
                                            label="Phone Number"
                                            name="phone_number"
                                            value={values.phone_number}
                                            handleChange={handleChange}
                                            error={submitCount > 0 && !!errors.phone_number}
                                            helperText={submitCount > 0 ? errors.phone_number : null}
                                        />
                                    </Grid>
                                    <Grid item xs={6}>
                                        <KeyboardDatePicker	
                                            id="birthday-picker"	
                                            clearable	
                                            disableFuture	
                                            format="MM/dd/yyyy"	
                                            placeholder="mm/dd/yyyy"	
                                            label="Birthday"	
                                            value={values.birthday || null}	
                                            fullWidth	
                                            onBlur={handleBlur}	
                                            onChange={e => setFieldValue('birthday', e)}	
                                            KeyboardButtonProps={{	
                                                'aria-label': 'change date',	
                                            }}	
                                        />	
                                    </Grid>
                                </Grid>
                                <ActionButton disabled={!allValuesSet(values) || isSubmitting} marginTop={20} marginBottom={20}>Save Changes</ActionButton>
                            </div>
                        </form>
                    )}
                </Formik>
            </>
        );
    }
}

AccountPage.propTypes = {
    updateApplicant: PropTypes.func,
}

const mapStateToProps = (state) => ({
    applicant: state.applicant,
});

const mapDispatchToProps = { updateApplicant };

export default connect(mapStateToProps, mapDispatchToProps)(captureRoute(AccountPage, ROUTES.ACCOUNT));
