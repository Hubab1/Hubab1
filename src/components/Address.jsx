import React, { useState, useCallback, useMemo } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { TextField } from 'formik-material-ui';
import { Formik, Form, Field } from 'formik';
import * as Yup from 'yup';
import styled from '@emotion/styled';

import { ROUTES } from 'app/constants';
import { updateApplicant } from 'reducers/applicant';
import withRelativeRoutes from 'app/withRelativeRoutes';
import ActionButton from 'components/common/ActionButton/ActionButton';
import LocationSearch from 'components/common/Formik/LocationSearch/LocationSearch';
import { H1, SpacedH3 } from 'assets/styles';
import sticky from 'assets/images/sticky.png';
import GenericFormMessage from './common/GenericFormMessage';

const ImageContainer = styled.div`
    margin-top: 31px;
    margin-bottom: 31px;
    img {
        max-height: 105px;
        max-width: 114px;
    }
`;

export const validationSchema = Yup.object().shape({
    address_search: Yup.object().shape({
        address_street: Yup.string()
            .required('Street is required')
            .matches(/^[A-Za-z0-9]+(?:\s[A-Za-z0-9'_-]+)+$/, 'Invalid street'),
        address_city: Yup.string()
            .required('City is required')
            .matches(/^[\w'\-,.][^0-9_!¡?÷?¿/\\+=@#$%ˆ&*(){}|~<>;:[\]]{1,}$/, 'Invalid city'),
        address_state: Yup.string().required('State is required'),
        address_postal_code: Yup.number().required('Zip code is required'),
    }),
    address_line_2: Yup.string(),
});

export const Address = ({ applicant, updateApplicant, _nextRoute }) => {
    const [errors, setErrors] = useState(null);
    const handleSubmit = useCallback(
        async (values, { setSubmitting, setErrors: setFormErrors }) => {
            try {
                const serialized = {
                    ...values.address_search,
                    address_line_2: values.address_line_2,
                };

                const response = await updateApplicant(serialized);
                if (response.errors) {
                    setFormErrors(response.errors);
                } else {
                    _nextRoute();
                }
            } catch {
                setErrors(['Oops! We ran into some issues. Please try again later.']);
            } finally {
                setSubmitting(false);
            }
        },
        [updateApplicant, _nextRoute]
    );

    const initialValues = useMemo(() => {
        const searchBuilder = [];
        if (applicant.address_street) searchBuilder.push(applicant.address_street);
        if (applicant.address_city) searchBuilder.push(applicant.address_city);
        if (applicant.address_state) searchBuilder.push(applicant.address_state);
        if (applicant.address_postal_code) searchBuilder.push(applicant.address_postal_code);
        const search = searchBuilder.join(', ');

        return {
            address_search: {
                search,
                address_street: applicant.address_street,
                address_city: applicant.address_city,
                address_state: applicant.address_state,
                address_postal_code: applicant.address_postal_code,
            },
            address_line_2: applicant.address_line_2,
        };
    }, [applicant]);

    if (!applicant) {
        return null;
    }

    return (
        <>
            <H1>Tell Us A Little More</H1>
            <SpacedH3>Now, by filling out these details below we can screen you more accurately.</SpacedH3>
            <ImageContainer>
                <img src={sticky} alt="sticky note" />
            </ImageContainer>
            <Formik validationSchema={validationSchema} initialValues={initialValues} onSubmit={handleSubmit}>
                {({ values, isSubmitting, handleSubmit, submitCount }) => {
                    const disableSubmit = !values.address_search.search || isSubmitting;

                    return (
                        <Form onSubmit={handleSubmit} autoComplete="off">
                            <Field
                                fullWidth
                                margin="normal"
                                label="Street name, city, state, zip"
                                name="address_search"
                                submitCount={submitCount}
                                component={LocationSearch}
                            />
                            <Field
                                fullWidth
                                margin="normal"
                                label="Apt/Ste/Floor"
                                name="address_line_2"
                                component={TextField}
                            />
                            <GenericFormMessage type="error" messages={errors} />
                            <ActionButton marginTop={50} disabled={disableSubmit}>
                                Continue
                            </ActionButton>
                        </Form>
                    );
                }}
            </Formik>
        </>
    );
};

Address.propTypes = {
    updateApplicant: PropTypes.func.isRequired,
    applicant: PropTypes.object,
    _nextRoute: PropTypes.func,
};

const mapStateToProps = (state) => ({
    applicant: state.applicant,
});

const mapActionsToProps = {
    updateApplicant,
};

export default connect(mapStateToProps, mapActionsToProps)(withRelativeRoutes(Address, ROUTES.ADDRESS));
