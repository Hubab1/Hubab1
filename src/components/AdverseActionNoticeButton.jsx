import React, { useState } from 'react';
import API from 'app/api';
import { LinkButton } from 'assets/styles';
import { css } from 'emotion';

export const aanLink = css`
    outline: none;
`;

export function AdverseActionNoticeButton() {
    const [error, setError] = useState(false);

    const openDocument = async () => {
        const w = window.open('data:text/plain;charset=utf-8;base64,TG9hZGluZw==', '_blank');
        try {
            const response = await API.fetchAANDocument();
            const blobUrl = URL.createObjectURL(new Blob([response], { type: 'application/pdf' }));
            w.location.replace(blobUrl);
        } catch (e) {
            setError(true);
        }
    };

    return (
        <LinkButton onClick={openDocument} className={aanLink}>
            {error ? 'An error occurred. Please try again.' : 'Learn why'}
        </LinkButton>
    );
}
