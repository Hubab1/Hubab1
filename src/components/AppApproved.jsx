import React from 'react';
import HelloSign from 'hellosign-embedded';
import styled from '@emotion/styled';
import Grid from '@material-ui/core/Grid';
import { css } from 'emotion';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import API from 'app/api';
import { ROUTES, HELLOSIGN_TEST_MODE, HELLOSIGN_CLIENT_ID, MILESTONE_LEASE_SENT } from 'app/constants';
import withRelativeRoutes from 'app/withRelativeRoutes';
import approvedSign from 'assets/images/approvedSign.svg';
import { H1, leftText, SpacedH3 } from 'assets/styles';
import ActionButton from 'components/common/ActionButton/ActionButton';
import lightbulb from 'assets/images/lightbulb.png';
import { prettyCurrency } from 'utils/misc';

export const ApprovedImage = styled.img`
    padding-top: 10px;
    height: 200px;
`;

export const applicationUnit = css`
  color: #454B57;
  font-size: 14px;
  line-height: 17px;
  text-align: center;
  padding-top: 10px;
`;

export const BulbImage = styled.img`
  width: 46px;
  height: 42px;
`;

export const securityDepositHelpText = css`
  color: #454B57;
  font-size: 14px;
  line-height: 17px;
  text-align: left;
`;

export const gridContainer = css`
    padding-top: 35px;
    min-height: 100px;
`;

export const AppApproved = ({profile, configuration, history}) => {
    if (!profile || ! configuration) return null;

    const {
        unit,
        security_deposit: securityDeposit,
        security_deposit_multiplier: securityDepositMultiplier,
    } = profile;
    const buildingName = configuration.community.building_name || configuration.community.normalized_street_address;
    const unitNumber = (!!unit && !!unit.unit_number) ? ` Unit ${unit.unit_number}` : '';

    const openEmbeddedSigning = async () => {
        const client = new HelloSign({
            clientId: HELLOSIGN_CLIENT_ID,
        });
        const data = await API.embeddedSigningUrl();
        if (data.url) {
            client && client.open(data.url, {
                testMode: HELLOSIGN_TEST_MODE,
                skipDomainVerification: HELLOSIGN_TEST_MODE,
            });
        }
        client.on('sign', () => {
            history.push(ROUTES.LEASE_SIGNED);
        });
    }
    const leaseSent = !!profile.events.find(e => String(e.event) === String(MILESTONE_LEASE_SENT));

    return (
        <>
            <H1>You've Been Approved!</H1>
            {
                leaseSent && <SpacedH3>All that's left to do is sign the lease.</SpacedH3>
            }
            {
                !leaseSent && <SpacedH3>We'll send an email with instructions on how to sign the lease shortly.</SpacedH3>
            }
            <ApprovedImage src={approvedSign}/>
            <div id="application-unit" className={applicationUnit}>{buildingName}{unitNumber}</div>
            <div className={gridContainer}>
                {securityDeposit &&
                <Grid container justify={'center'} className="security-deposit-container">
                    <Grid item xs={2}>
                        <BulbImage alt="light bulb" src={lightbulb} />
                    </Grid>
                    <Grid item xs={9} classes={{ root: leftText }}>
                        <span className={securityDepositHelpText}>
                            You've been approved under the condition that you agree to
                            a {prettyCurrency(securityDeposit)} security deposit
                            ({securityDepositMultiplier}x the rent).
                        </span>
                    </Grid>
                </Grid>}
                {
                    leaseSent &&
                    <ActionButton onClick={openEmbeddedSigning} marginTop={securityDeposit ? 30 : 90}>
                        Review &amp; Sign Lease
                    </ActionButton>
                }
            </div>
        </>
    )
};

AppApproved.propTypes = {
    profile: PropTypes.object,
    configuration: PropTypes.object,
};


const mapStateToProps = state => ({
    profile: state.renterProfile,
    configuration: state.configuration,
});

export default connect(mapStateToProps)(withRelativeRoutes(AppApproved, ROUTES.APP_APPROVED));
