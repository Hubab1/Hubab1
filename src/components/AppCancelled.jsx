import React from 'react';
import styled from '@emotion/styled';
import { css } from 'emotion';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { ROUTES } from 'app/constants';
import { H1, SpacedH3, P } from 'assets/styles';
import cancelled from 'assets/images/cancelled.png';
import { prettyFormatPhoneNumber } from 'utils/misc';
import captureRoute from 'app/captureRoute';

export const Img = styled.img`
    padding-top: 10px;
    height: 200px;
`;

export const applicationUnit = css`
  color: #454B57;
  font-size: 14px;
  line-height: 17px;
  text-align: center;
  padding-top: 10px;
`;

export const AppCancelled = ({profile, configuration}) => {
    if (!profile || ! configuration) return null;

    return (
        <>
            <H1>Application Cancelled</H1>
            <SpacedH3>You are no longer applying for unit {profile.unit.unit_number}.</SpacedH3>
            <Img src={cancelled}/>
            <P margin="90px 0 0 0" bold>Call us at <a href={`tel:${configuration.community.contact_phone}`}>{prettyFormatPhoneNumber(configuration.community.contact_phone)}</a>
                &nbsp;if you have any questions or if youâ€™d like to start a new application.
            </P>
        </>
    )
};

AppCancelled.propTypes = {
    profile: PropTypes.object,
    configuration: PropTypes.object,
};


const mapStateToProps = state => ({
    profile: state.renterProfile,
    configuration: state.configuration,
});

export default connect(mapStateToProps)(captureRoute(AppCancelled, ROUTES.APP_CANCELLED));
