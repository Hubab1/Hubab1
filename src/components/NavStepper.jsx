import React from 'react';
import { connect } from 'react-redux';
import { css } from 'emotion';
import { makeStyles } from '@material-ui/core/styles';
import { withRouter } from 'react-router-dom';
import Stepper from '@material-ui/core/Stepper';
import Step from '@material-ui/core/Step';
import StepLabel from '@material-ui/core/StepLabel';
import clsx from 'clsx';
import ErrorIcon from '@material-ui/icons/Error';
import PropTypes from 'prop-types';
import { MOCKY } from 'app/api';
import { selectors } from 'reducers/renter-profile';
import { actions } from 'reducers/store';
import { prettyFormatPhoneNumber } from 'utils/misc';
import Button from '@material-ui/core/Button';
import { ROUTES } from 'app/constants';

const useStyles = makeStyles(() => ({
    root: {
        width: '100%',
    },
}));

const iconRoot = css`
    align-items: flex-start !important;
    .appCompletedMsg,
    .outstandingBalance {
        color: #828796;
    }
`;

const active = css`
    .MuiStepLabel-active {
        font-weight: bold !important;
    }
`;

const accessible = css`
    cursor: pointer !important;
    .Mui-disabled {
        cursor: pointer !important;
    }
`;

const viewProgress = css`
    width: 192px;
    border-radius: 30px !important;
    text-transform: none !important;
    font-size: 16px !important;
    margin: 30px auto !important;
    display: block !important;
`;

export function getStepperIndex(routes, currentRoute) {
    for (let i = 0; i < routes.length; i++) {
        const route = routes[i];
        if (route.value === currentRoute) return i;
    }
    return -1;
}

export function VerticalLinearStepper(props) {
    const classes = useStyles();

    const activeStep = getStepperIndex(props.navRoutes, props.currentRoute);
    const firstUncompletedStep = getStepperIndex(props.navRoutes, props.defaultInitialPage);
    const unitUnavailable = props.renterProfile?.unit_available === false;
    const outstandingBalance = props.initialPage === ROUTES.OUTSTANDING_BALANCE;
    const holdingDepositAgreementSignatureRequested = props.initialPage === ROUTES.HOLDING_DEPOSIT_TERMS_AGREEMENT;

    function onClickRoute(e, route, i) {
        e.stopPropagation();
        if (i <= firstUncompletedStep || MOCKY) {
            props.history.push(route.value);
        }
    }

    return (
        <div className={classes.root}>
            <Stepper activeStep={activeStep} orientation="vertical">
                {unitUnavailable && !props.guarantorRequested && (
                    <Step active>
                        <StepLabel
                            StepIconComponent={() => <ErrorIcon color="primary" />}
                            active
                            classes={{ root: iconRoot }}
                        >
                            <span className="unitUnavailableMsg">
                                {`We've placed your application on hold for now, since the apartment you were interested in is no longer available. Please call us at `}
                                <a href={`tel:${props.config.community.contact_phone}`}>
                                    {prettyFormatPhoneNumber(props.config.community.contact_phone)}
                                </a>{' '}
                                so we can discuss some other options.
                            </span>
                        </StepLabel>
                        <Button
                            variant="outlined"
                            color="default"
                            id="viewProgressButton"
                            classes={{
                                root: viewProgress,
                            }}
                            disabled={false}
                            onClick={() => props.history.push(props.initialPage)}
                        >
                            View Progress
                        </Button>
                    </Step>
                )}
                {holdingDepositAgreementSignatureRequested && (
                    <Step active>
                        <StepLabel
                            StepIconComponent={() => <ErrorIcon color="primary" />}
                            active
                            classes={{ root: iconRoot }}
                        >
                            <span className="holdingDepositReagreement">
                                We’ll need you to agree to the new holding deposit terms. Please call us at&nbsp;
                                <a href={`tel:${props.config.community.contact_phone}`}>
                                    {prettyFormatPhoneNumber(props.config.community.contact_phone)}
                                </a>{' '}
                                if you have any questions.
                            </span>
                        </StepLabel>
                        <Button
                            variant="outlined"
                            color="primary"
                            id="viewProgressButton"
                            classes={{
                                root: viewProgress,
                            }}
                            disabled={false}
                            onClick={() => {
                                props.history.push(props.initialPage);
                                props.handleDrawerClose();
                            }}
                        >
                            View Progress
                        </Button>
                    </Step>
                )}
                {props.applicantStillFinishingApplication &&
                    !unitUnavailable &&
                    !props.guarantorRequested &&
                    !holdingDepositAgreementSignatureRequested &&
                    props.navRoutes.map((route, i) => (
                        <Step
                            classes={{
                                root: clsx({
                                    [active]: activeStep === i,
                                    [accessible]: i <= firstUncompletedStep,
                                }),
                            }}
                            key={route.name}
                            onClick={(e) => onClickRoute(e, route, i)}
                            active={i === activeStep || i === firstUncompletedStep}
                        >
                            <StepLabel icon={' '} completed={i < firstUncompletedStep}>
                                {route.name}
                            </StepLabel>
                        </Step>
                    ))}
                {props.guarantorRequested && (
                    <Step active>
                        <StepLabel
                            StepIconComponent={() => <ErrorIcon color="primary" />}
                            active
                            classes={{ root: iconRoot }}
                        >
                            <span className="appCompletedMsg">
                                We’re waiting for you to add a guarantor. Please call us at&nbsp;
                                <a href={`tel:${props.config.community.contact_phone}`}>
                                    {prettyFormatPhoneNumber(props.config.community.contact_phone)}
                                </a>{' '}
                                if you have any questions or if you are unable or unwilling to add a guarantor.
                            </span>
                        </StepLabel>
                        <Button
                            variant="outlined"
                            color="primary"
                            id="viewProgressButton"
                            classes={{
                                root: viewProgress,
                            }}
                            disabled={false}
                            onClick={() => props.history.push(props.initialPage)}
                        >
                            View Progress
                        </Button>
                    </Step>
                )}
                {!props.applicantStillFinishingApplication &&
                    !props.guarantorRequested &&
                    !outstandingBalance &&
                    !holdingDepositAgreementSignatureRequested && (
                        <Step active>
                            <StepLabel completed classes={{ root: iconRoot }}>
                                <span className="appCompletedMsg">
                                    Your application has been completed and submitted. Please call us at&nbsp;
                                    <a href={`tel:${props.config.community.contact_phone}`}>
                                        {prettyFormatPhoneNumber(props.config.community.contact_phone)}
                                    </a>{' '}
                                    if you have any questions.
                                </span>
                            </StepLabel>
                            <Button
                                variant="outlined"
                                color="primary"
                                id="viewProgressButton"
                                classes={{
                                    root: viewProgress,
                                }}
                                disabled={false}
                                onClick={() => {
                                    props.history.push(props.initialPage);
                                    props.handleDrawerClose();
                                }}
                            >
                                View Progress
                            </Button>
                        </Step>
                    )}
                {outstandingBalance && (
                    <Step active>
                        <StepLabel
                            labelContainer
                            classes={{ root: iconRoot }}
                            StepIconComponent={() => <ErrorIcon color="primary" />}
                        >
                            <span className="outstandingBalance">
                                {
                                    "You'll be able to move forward with your application once all outstanding balances have been paid. Please call us at "
                                }
                                <a href={`tel:${props.config.community.contact_phone}`}>
                                    {prettyFormatPhoneNumber(props.config.community.contact_phone)}
                                </a>{' '}
                                if you have any questions.
                            </span>
                        </StepLabel>
                        <Button
                            variant="outlined"
                            color="primary"
                            id="viewProgressButton"
                            classes={{
                                root: viewProgress,
                            }}
                            disabled={false}
                            onClick={() => {
                                props.history.push(ROUTES.OUTSTANDING_BALANCE);
                                props.handleDrawerClose();
                            }}
                        >
                            View Progress
                        </Button>
                    </Step>
                )}
            </Stepper>
        </div>
    );
}

VerticalLinearStepper.propTypes = {
    navRoutes: PropTypes.array,
    currentRoute: PropTypes.string,
    initialPage: PropTypes.string,
    defaultInitialPage: PropTypes.string,
    renterProfile: PropTypes.object,
    history: PropTypes.object,
    config: PropTypes.object,
    applicantStillFinishingApplication: PropTypes.bool,
    guarantorRequested: PropTypes.bool,
    handleDrawerClose: PropTypes.func.isRequired,
};

const mapDispatchToProps = {
    logout: actions.logout,
};

const mapStateToProps = (state) => ({
    navRoutes: selectors.selectNav(state),
    currentRoute: state.siteConfig.currentRoute,
    initialPage: selectors.selectInitialPage(state),
    defaultInitialPage: selectors.selectDefaultInitialPage(state),
    applicantStillFinishingApplication: selectors.selectApplicantStillFinishingApplication(state),
    guarantorRequested: selectors.selectGuarantorRequested(state),
    renterProfile: state.renterProfile,
    config: state.configuration,
});

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(VerticalLinearStepper));
