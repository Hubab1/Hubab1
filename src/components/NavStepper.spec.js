import React from 'react';
import {shallow} from 'enzyme';
import { getStepperIndex, VerticalLinearStepper } from 'components/NavStepper';
import { ROUTES } from 'app/constants';

describe('getStepperIndex', function(){
    it('gets the correct index for an unnested route', function() {
        const routes = [
            {name: 'Current Address', value: ROUTES.ADDRESS},
            {name: 'Lease Terms', value: ROUTES.LEASE_TERMS},
            {name: 'Income & Employment', value: ROUTES.INCOME_AND_EMPLOYMENT},
            {name: 'Screening', value: ROUTES.SCREENING},
        ];
        
        const currentRoute = ROUTES.INCOME_AND_EMPLOYMENT;
        expect(getStepperIndex(routes, currentRoute)).toEqual(2);
    });

    it('returns -1 if route not found', function() {
        const routes = [
            {name: 'Lease Terms', value: ROUTES.LEASE_TERMS},
            {name: 'Rental Profile', value: ROUTES.PROFILE_OPTIONS},
            {name: 'Screening', value: ROUTES.SCREENING},
        ];
        
        const currentRoute = 'FAKEROUTE';
        expect(getStepperIndex(routes, currentRoute)).toEqual(-1);
    });
});

describe('VerticalLinearStepper', () => {
    const defaultProps = {
        initialPage: 'income',
        currentRoute: 'lease_settings',
        applicantStillFinishingApplication: true,
        navRoutes: [
            {
                name: 'lease settings',
                value: 'lease_settings',
            },
            {
                name: 'profile',
                value: 'profile',
            },
            {
                name: 'income',
                value: 'income',
            },
            {
                name: 'fees',
                value: 'fees',
            },
            {
                name: 'screening',
                value: 'screening',
            }
        ],
        config: {
            community: {
                contact_phone: '123-456-7891'
            }
        }
    };
    it('matches snapshot', () => {
        const wrapper = shallow(<VerticalLinearStepper {...defaultProps}/>);
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});

describe('Application submitted state', function() {
    it('renders an application already submited message', function() {
        const defaultProps = {
            navRoutes: [],
            config: {
                community: {
                    contact_phone: '123-456-7891'
                }
            }
        };
        const wrapper = shallow(<VerticalLinearStepper {...defaultProps} />);
        const appCompletedMsg =  wrapper.find('.appCompletedMsg');
        expect(appCompletedMsg.text()).toContain('Your application has been completed and submitted. Please call us at 123‑456‑7891 if you have any questions.');
        expect(wrapper.find('#viewProgressButton').text()).toContain('View Progress');
    });

    it('View Progress when clicked takes to the initialPage set', function () {
        const defaultProps = {
            navRoutes: [],
            config: {
                community: {
                    contact_phone: '123-456-7891'
                }
            },
            history: {
                push: jest.fn(),
            },
            initialPage: '/application-complete',
        };
        let wrapper = shallow(<VerticalLinearStepper {...defaultProps} />);
        wrapper.find('#viewProgressButton').simulate('click');
        expect(defaultProps.history.push).toHaveBeenCalledWith('/application-complete');
    });
});

describe('Unit unavailable state', function() {
    it('Renders a unit unavailable message', function() {
        const defaultProps = {
            renterProfile: {
                unit_available: false,
            },
            applicantStillFinishingApplication: true,
            navRoutes: [],
            config: {
                community: {
                    contact_phone: '123-456-7891'
                }
            }
        };
        const wrapper = shallow(<VerticalLinearStepper {...defaultProps} />);
        const appCompletedMsg =  wrapper.find('.unitUnavailableMsg');
        expect(appCompletedMsg.text()).toContain("We've placed your application on hold for now, since the apartment you were interested in is no longer available. Please call us at 123‑456‑7891 so we can discuss some other options.");
        expect(wrapper.find('#viewProgressButton').text()).toContain('View Progress');
    });

    it('View Progress when clicked takes to initialPage set', function() {
        const defaultProps = {
            renterProfile: {
                unit_available: false,
            },
            applicantStillFinishingApplication: true,
            navRoutes: [],
            config: {
                community: {
                    contact_phone: '123-456-7891'
                }
            },
            history: {
                push: jest.fn(),
            },
            initialPage: ROUTES.UNIT_UNAVAILABLE,
        };
        const wrapper = shallow(<VerticalLinearStepper {...defaultProps} />);
        wrapper.find('#viewProgressButton').simulate('click');
        expect(defaultProps.history.push).toHaveBeenCalledWith('/unit-unavailable');
    });
});

describe('Guarantor requested state', function() {
    it('renders request guarantor message', function() {
        const defaultProps = {
            navRoutes: [],
            config: {
                community: {
                    contact_phone: '123-456-7891'
                }
            },
            initialPage: '/guarantor_request',
            guarantorRequested: true,
        };
        const wrapper = shallow(<VerticalLinearStepper {...defaultProps} />);
        const appCompletedMsg =  wrapper.find('.appCompletedMsg');
        expect(appCompletedMsg.text()).toContain('We’re waiting for you to add a guarantor. Please call us at 123‑456‑7891 if you have any questions or if you are unable or unwilling to add a guarantor.');
        expect(wrapper.find('#viewProgressButton').text()).toContain('View Progress');
    });

    it('View Progress when clicked takes to the initialPage set', function () {
        const defaultProps = {
            navRoutes: [],
            config: {
                community: {
                    contact_phone: '123-456-7891'
                }
            },
            history: {
                push: jest.fn(),
            },
            initialPage: '/guarantor_request',
            guarantorRequested: true,
        };
        let wrapper = shallow(<VerticalLinearStepper {...defaultProps} />);
        wrapper.find('#viewProgressButton').simulate('click');
        expect(defaultProps.history.push).toHaveBeenCalledWith('/guarantor_request');
    });
});

