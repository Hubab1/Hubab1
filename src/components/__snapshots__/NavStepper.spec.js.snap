// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerticalLinearStepper matches snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <ForwardRef(WithStyles)
    activeStep={0}
    orientation="vertical"
  >
    <ForwardRef(WithStyles)
      active={true}
      classes={
        Object {
          "root": "css-1wi6ago",
        }
      }
      onClick={[Function]}
    >
      <ForwardRef(WithStyles)
        completed={false}
        icon=" "
      >
        lease settings
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)
      active={false}
      classes={
        Object {
          "root": "",
        }
      }
      onClick={[Function]}
    >
      <ForwardRef(WithStyles)
        completed={false}
        icon=" "
      >
        profile
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)
      active={false}
      classes={
        Object {
          "root": "",
        }
      }
      onClick={[Function]}
    >
      <ForwardRef(WithStyles)
        completed={false}
        icon=" "
      >
        income
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)
      active={false}
      classes={
        Object {
          "root": "",
        }
      }
      onClick={[Function]}
    >
      <ForwardRef(WithStyles)
        completed={false}
        icon=" "
      >
        fees
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)
      active={false}
      classes={
        Object {
          "root": "",
        }
      }
      onClick={[Function]}
    >
      <ForwardRef(WithStyles)
        completed={false}
        icon=" "
      >
        screening
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
  </ForwardRef(WithStyles)>
</div>
`;
